### PR TITLE
Ubuntu 24.04 CIS section 6.1.2.1.3 Ensure systemd-journal-upload is enabled and active

### PR DESCRIPTION
**Pull request by ericeberry**
#### Description:

Ubuntu 24.04 CIS section 6.1.2.1.3 Ensure systemd-journal-upload is enabled and active

#### Rationale:

- Implement Ensure systemd-journal-upload is enabled and active for Ubuntu 24.04

### CODE REVIEW

File: linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_disabled.fail.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (mpurg) [inline linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_disabled.fail.sh:1]: This test is not needed. The rule is using a template which already has this test defined:
- shared/templates/service_enabled/tests/service_disabled.fail.sh

File: linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_disabled.fail.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+# packages = systemd
+
+{{{ bash_package_install("systemd-journal-upload") }}}
```
Comment (mpurg) [inline linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_disabled.fail.sh:4]: The package name should be `systemd-journal-remote`

File: linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_enabled.pass.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (mpurg) [inline linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_enabled.pass.sh:1]: This test is not needed. The rule is using a template which already has this test defined:
- shared/templates/service_enabled/tests/service_enabled.pass.sh

File: products/ubuntu2004/product.yml [hunk 4]
```diff
@@ -46,6 +46,7 @@ platform_package_overrides:
   pam: libpam-runtime
   shadow: login
   sssd: sssd-common
+  systemd-journal-upload: systemd-journal-remote
```
Comment (mpurg) [inline products/ubuntu2004/product.yml:49]: I'm not sure if this change is needed.
- `systemd-journal-upload` is the name of the service, not a package that would need overriding. I also don't think the package needs overriding.
- this PR is about Ubuntu 24.04, and this change is on Ubuntu 20.04 product

File: tests/data/product_stability/ubuntu2404.yml [hunk 5]
```diff
@@ -60,6 +60,7 @@ platform_package_overrides:
   shadow: login
   sssd: sssd-common
   sssd-ldap: null
+  systemd-journal-upload: systemd-journal-remote
```
Comment (mpurg) [inline tests/data/product_stability/ubuntu2404.yml:63]: I'm not sure if this is needed, see previous comment.

**Review by mpurg — CHANGES_REQUESTED** [review]
Few changes needed

File: products/ubuntu2004/product.yml (hunk 4 above)
Comment (ericeberry) [reply products/ubuntu2004/product.yml:49]: Removed

File: linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_enabled.pass.sh (hunk 3 above)
Comment (ericeberry) [reply linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_enabled.pass.sh:1]: Removed

File: linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_disabled.fail.sh (hunk 1 above)
Comment (ericeberry) [reply linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_disabled.fail.sh:1]: Removed

File: tests/data/product_stability/ubuntu2404.yml (hunk 5 above)
Comment (ericeberry) [reply tests/data/product_stability/ubuntu2404.yml:63]: Removed

File: linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_disabled.fail.sh (hunk 2 above)
Comment (ericeberry) [reply linux_os/guide/system/logging/journald/service_systemd-journal-upload_enabled/tests/service_disabled.fail.sh]: File removed

File: controls/cis_ubuntu2404.yml [hunk 6]
```diff
@@ -2261,7 +2261,8 @@ controls:
       - l1_server
       - l1_workstation
     status: planned
```
Comment (mpurg) [inline controls/cis_ubuntu2404.yml:2263]: status should be changed to automated

**Review by mpurg — CHANGES_REQUESTED** [review]
minor change

**Review by dodys — APPROVED** [review]
lgtm, thanks!

**Pull request merged**